typescript-test-samples/launchdarkly-lambda-dynamodb/readme.md

```diff
@@ -0,0 +1,90 @@
+
+For a complete guide to using LaunchDarkly in AWS serverless, including how to set up the Relay Proxy within your AWS environment and how to ensure that all events are captured by the SDKs within an ephemeral Lambda, visit [this guide](https://launchdarkly.com/blog/using-launchdarkly-in-aws-serverless/).
+
+The project also includes the code necessary to build and deploy a Lambda Layer to easily add the necessary LaunchDarkly dependencies to any Lambda that requires them. More on how to create and use the Layer below.
+
+## About This Example
+
+It can be difficult to fully test features in Lambdas outside of a production cloud environment. Feature flags can be a powerful pattern for enabling and disabling features for testing. LaunchDarkly's advanced targeting capabilities allow you to create feature flags that canbe used to safely test features in production without impacting the existing experience. Using feature flags and targeting, you can run code paths that are enabled for one user or one environment (or any combination of criteria) but not for anyone else. For example, utilizing authentication, environment and identity data, you can enable features for:
+
+* Only particular testing or staging environments
+* Only particular users or segments of users
+* Beta testing with managed beta access customers
+* Progressive rollouts to stress testing features
+* Experimentation on features or performance
+
+...and much more. 
+
+## The LaunchDarkly SDKs
+
+Adding LaunchDarkly feature flags to any Lambda always begins with installing the appropriate SDK for the Lambda runtime you are using. You can find a [full list of server-side SDKs here](https://docs.launchdarkly.com/sdk). For the Node.js runtime, that would mean using NPM.
+
+```bash
+npm install launchdarkly-node-server-sdk
+```
+
+If you would like to use [DynamoDB as a persistent feature store](https://docs.launchdarkly.com/sdk/features/storing-data/dynamodb/?q=dynamo), you'll need to also install those dependencies.
+
+```bash
+npm install launchdarkly-node-server-sdk-dynamodb
+```
+
+This works well for a single Lambda, but if you are using feature flags across a number of Lambdas, the easiest method to maintain the depencies across all of them is using a Lambda Layer. The code provided here will assist you in creating the Layer.
+
+### Creating/Updating the Lambda Layer
+
+The necessary dependencies are already included in this project under `/cdk/resources/layers`. It contains both the Node.js SDK and the DynamoDB extension. In order to ensure that this asset has the latest versions of these SDKs, we've included a script that will download the latest versions and archive them into the proper folder. It is recommended that you run this before deploying any of the assets using the provided CDK scripts. All you need to do is:
```

Comment:
Thanks for including this script. Looks like there's a `layer.zip` included in the repo under `resources/layers`. Can we drop that since users will be creating the layer itself (thinking of managing overall repo size here).